src/devices/cpu/uml.h

```diff
@@ -577,10 +577,10 @@ namespace uml
 	private:
 		// internal configuration
 		void configure(opcode_t op, u8 size, condition_t cond = COND_ALWAYS);
-		void configure(opcode_t op, u8 size, parameter p0, condition_t cond = COND_ALWAYS);
-		void configure(opcode_t op, u8 size, parameter p0, parameter p1, condition_t cond = COND_ALWAYS);
-		void configure(opcode_t op, u8 size, parameter p0, parameter p1, parameter p2, condition_t cond = COND_ALWAYS);
-		void configure(opcode_t op, u8 size, parameter p0, parameter p1, parameter p2, parameter p3, condition_t cond = COND_ALWAYS);
+		void configure(opcode_t op, u8 size, const parameter &p0, condition_t cond = COND_ALWAYS);
+		void configure(opcode_t op, u8 size, const parameter &p0, const parameter &p1, condition_t cond = COND_ALWAYS);
+		void configure(opcode_t op, u8 size, const parameter &p0, const parameter &p1, const parameter &p2, condition_t cond = COND_ALWAYS);
+		void configure(opcode_t op, u8 size, const parameter &p0, const parameter &p1, const parameter &p2, const parameter &p3, condition_t cond = COND_ALWAYS);
```

Comment:
`parameter` is small small enough to be passed by value in x64 ABI (it's an `int`-sized enum and a u64), and it has a trivial destructor.  Changing the code to pass `parameter` by reference like this hurts locality and makes it slower when passing a temporary because the value needs to be stored elsewhere in the frame and a reference to it stacked.

Reply:
That's what I assumed, but I wanted to get feedback on this.

It's passed multiple times IIRC so I wasn't sure if it wouldn't benefit after all.

Clang is able to detect trivially-copyable objects and I think this should be configurable for that check.

Reply:
Maybe the `constexpr` throws off the heuristic. I will do some tests.

Reply:
The copy constructor implementation causes this warning. It seems to make the object not trivial.

This is enough to trigger the warnings
```
class parameter
{
public:
	parameter() { }
	parameter(parameter const &param) { }
};
```

Ironically clang-tidy also reports that the copy constructor should be made a trivial one by using `= default` for it.

Reply:
According to http://www.enseignement.polytechnique.fr/informatique/INF478/docs/Cpp/en/cpp/language/copy_constructor.html a user provided copy-constructor makes it non-trivial. No idea what actual effect it has on the code generation.

Reply:
Playing with godbolt - most combination generate the same code. Except for the copy-by-value case and "= default" which actually generates an additional instruction. As I have no clue about instructions I cannot say which code is better :P